src/main/java/gyro/google/dns/ManagedZoneResource.java

```diff
@@ -0,0 +1,405 @@
+/*
+ * Copyright 2019, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.google.dns;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import com.google.api.client.util.Data;
+import com.google.api.services.dns.Dns;
+import com.google.api.services.dns.model.ManagedZone;
+import com.google.api.services.dns.model.ManagedZoneDnsSecConfig;
+import com.google.api.services.dns.model.ManagedZoneForwardingConfig;
+import com.google.api.services.dns.model.ManagedZonePrivateVisibilityConfig;
+import com.google.api.services.dns.model.Operation;
+import gyro.core.GyroException;
+import gyro.core.GyroUI;
+import gyro.core.Type;
+import gyro.core.resource.Output;
+import gyro.core.resource.Resource;
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.ConflictsWith;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.core.validation.ValidationError;
+import gyro.google.Copyable;
+import gyro.google.GoogleResource;
+
+/**
+ * Creates a Managed Zone.
+ *
+ * Example
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *     google::dns-managed-zone managed-zone-example-private
+ *         name: "managed-zone-example-private"
+ *         description: "Private Managed Zone Example"
+ *         dns-name: "private.example.com."
+ *         visibility: "private"
+ *
+ *         private-visibility-config
+ *             network
+ *                 network: $(google::compute-network network-example)
+ *             end
+ *         end
+ *     end
+ *
+ *     google::dns-managed-zone managed-zone-example-public
+ *         name: "managed-zone-example-public"
+ *         description: "Public Managed Zone Example"
+ *         dns-name: "p.example.com."
+ *         dnssec-config
+ *             state: "on"
+ *         end
+ *     end
+ */
+@Type("dns-managed-zone")
+public class ManagedZoneResource extends GoogleResource implements Copyable<ManagedZone> {
+
+    private String description;
+
```

Comment:
I don't think the blank lines between each field are consistent with our code style.

Reply:
Code style should be enforced by `checkstyle.xml` or `intellij-code-style.xml`. If it can't be done by them, a rule should be written down somewhere so that everyone can follow it.
but honestly I wouldn't be bothered if it can't be automated since there's no way we can enforce them 100%.